ros2_foxglove_bridge/src/message_definition_cache.cpp

```diff
@@ -133,6 +133,21 @@ static std::tuple<std::string, std::string, std::string> split_action_msg_defini
   return {definitions[0], definitions[1], definitions[2]};
 }
 
+/// @brief Split an service definition into individual request and response definitions.
+/// @param service_definition The full services definition as read from a .srv file
+/// @return A tuple holding request and response definitions
+static std::tuple<std::string, std::string> split_service_msg_definition(
+  const std::string& service_definition) {
+  constexpr char SEP[] = "---\n";
+
+  const auto definitions = split_string(service_definition, SEP);
+  if (definitions.size() != 2) {
```

Comment:
Is it legal for the response to be totally empty (no newline) e.g. something like `"int a\n---\nint b\n---"` ?

Reply:
Yes, I believe that's legal, also for the request. I will test if that's an issue

Reply:
"---\n" is correct:

> A service file contains two message definitions which are separated by a line which only contains three dashes:

Reply:
However, colcon doesn't raise an error if the service definition ends with `---` without final LF. I will have to cover that case as well :sob: